### ISSUE TITLE
The documentation has broken links from external sites

### ISSUE THREAD

**Issue by gaaclarke** [flutter/flutter#123272]

**Comment by a-siva** [comment]
@goderbauer what is the dart-dependency part in this ?

**Comment by goderbauer** [comment]
The dart-dependency here was: Can the DartDoc team help with this?

**Comment by a-siva** [comment]
> The dart-dependency here was: Can anything be done in the DartDoc tool to help with this?

@srawlins is this something that the Dart Doc tool could help with ?

**Comment by srawlins** [comment]
Yes, we could add any number of features to generate additional redirecting HTML files.

E.g. in a dartdoc_options.yaml file, could specify e.g.:

```yaml
redirects:
  - from: foundation/UniqueKey-class.html
  - to: widgets/UniqueKey-class.html
```

Or maybe an annotation on `UniqueKey` class, `@IUsedToLiveIn(['package:flutter/foundation.dart', ...])`.

If the flutter team is interesting in maintaining such a redirect list or set of annotations. As for tooling to automatically detect old locations and redirect to new locations, that would be outside dartdoc's purview.
